src/components/NumberInput/index.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import InputContainer, { InputContainerProps } from '../InputContainer';
+import RawInput, { RawInputProps } from '../RawInput';
+
+export type NumberInputProps<T> = Omit<InputContainerProps, 'input'> & RawInputProps<T>;
+
+function NumberInput<T extends string>(props: NumberInputProps<T>) {
```

Comment:
The value for number input should be either `number` or `undefined`

Reply:
T is for `name` not for value in this case

Reply:
Not talking about the T for `name` here!